packet.go

```diff
@@ -584,10 +591,13 @@ func (p *sshFxpReadPacket) UnmarshalBinary(b []byte) error {
 	return nil
 }
 
-func (p *sshFxpReadPacket) getDataSlice() []byte {
+func (p *sshFxpReadPacket) getDataSlice(alloc *allocator, orderID uint32) []byte {
 	dataLen := clamp(p.Len, maxTxPacket)
 	// we allocate a slice with a bigger capacity so we avoid a new allocation in sshFxpDataPacket.MarshalBinary
-	// and in sendPacket, we need 9 bytes in MarshalBinary and 4 bytes in sendPacket
+	// and in sendPacket, we need 9 bytes in MarshalBinary and 4 bytes in sendPacket.
+	if alloc != nil {
+		return alloc.GetPage(orderID)[:dataLen]
```

Comment:
Should `GetPage` also be returning a slice with capacity of `maxTxPacket+9+4`?

Reply:
GetPage returns a slice with capacity = maxMsgLength this is enough to avoid new allocations, I'll add a comment to clarify this